packages/time-input/src/utils/isSameUTCDayAndTime/isSameUTCDayAndTime.ts

```diff
@@ -0,0 +1,25 @@
+import { DateType, isValidDate } from '@leafygreen-ui/date-utils';
+
+/**
+ * Checks if two dates are the same day and time in UTC.
+ *
+ * @param day1 - The first date to check
+ * @param day2 - The second date to check
+ * @returns Whether the two dates are the same day and time in UTC
+ */
+export const isSameUTCDayAndTime = (
```

Comment:
We can probably use `isEqual` from `date-fns` 
https://date-fns.org/v4.1.0/docs/isEqual